crates/dojo-core/src/interfaces.cairo

```diff
@@ -15,14 +15,16 @@ trait IWorld<T> {
     fn register_component(ref self: T, class_hash: ClassHash);
     fn system(self: @T, name: felt252) -> ClassHash;
     fn register_system(ref self: T, class_hash: ClassHash);
-    fn uuid(self: @T) -> usize;
+    fn uuid(ref self: T) -> usize;
     fn emit(self: @T, keys: Span<felt252>, values: Span<felt252>);
-    fn execute(ref self: T, name: felt252, calldata: Span<felt252>) -> Span<felt252>;
-    fn entity(self: @T, component: felt252, key: Query, offset: u8, length: usize) -> Span<felt252>;
-    fn set_entity(
-        ref self: T, component: felt252, key: Query, offset: u8, value: Span<felt252>
-    );
-    fn entities(self: @T, component: felt252, partition: felt252) -> (Span<felt252>, Span<Span<felt252>>);
+    fn execute(ref self: T, system: felt252, calldata: Span<felt252>) -> Span<felt252>;
```

Comment:
As a follow up, it would be great to move these to the implementation's file

Reply:
I have open #588 @tarrencev to follow up this topic.